matRad_generateStf.m

```diff
@@ -313,36 +304,56 @@
                 
                 
             else % target not hit
-                stf(i).ray(j) = [];
-            end
-                        
+                stf(i).ray(j)       = [];
+           end
+                 
         end
 
         % book keeping
         stf(i).numOfRays = size(stf(i).ray,2);
         for j = 1:stf(i).numOfRays
+            
             stf(i).numOfBixelsPerRay(j) = numel(stf(i).ray(j).energy);
-            for k = 1:stf(i).numOfBixelsPerRay(j)
-                stf(i).ray(j).focusIx(k) = 
+            focusIx  =  machine.meta.defaultFociIndex * ones(stf(i).numOfBixelsPerRay(j),1);
+            [~, energyIx] = min(abs(bsxfun(@minus,[machine.data.energy]',...
+                            repmat(stf(i).ray(j).energy,length([machine.data]),1))));
+                   
+            % get for each spot the focus index
+            Cnt = 1; 
+            for k = energyIx
```

Comment:
change into loop over numOfBixelsPerRay to remove counter and arrive at a more clear structure of the code